python/MyriaPythonWorker.py

```diff
@@ -0,0 +1,218 @@
+            length = cls.read_int(stream)
+
+            if length == SpecialLengths.NULL or length == 0:
+                datalist.append(0)
+            # length is > 0, read the item now
+            elif length > 0:
+                obj = cls.read_item(stream, element_type, length)
+                datalist.append(obj)
+            else:
+                raise ValueError("Invalid length for item.")
+
+        return datalist
+
+    @classmethod
+    def write_with_length(cls, obj, stream, output_type):
+        if output_type == DataType.INT:
+            cls.write_int(DataType.INT, stream)
+            cls.write_int(obj, stream)
+        elif output_type == DataType.LONG:
+            cls.write_int(DataType.LONG, stream)
+            cls.write_long(obj, stream)
+        elif output_type == DataType.FLOAT:
+            cls.write_int(DataType.FLOAT, stream)
+            cls.write_float(obj, stream)
+        elif output_type == DataType.DOUBLE:
+            cls.write_int(DataType.DOUBLE, stream)
+            cls.write_double(obj, stream)
+        elif output_type == DataType.BLOB:
+            cls.write_int(DataType.BLOB, stream)
+            cls.pickle_and_write(obj, stream)
+
+    @classmethod
+    def read_command(cls, stream):
+        length = cls.read_int(stream)
+
+        if length < 0:
+            raise ValueError("Command cannot be less than zero.")
+        s = stream.read(length)
+
+        if len(s) < length:
+            raise EOFError
+        unenc = base64.urlsafe_b64decode(s)
+        return cls.loads(unenc)
+
+    @staticmethod
+    def dumps(obj):
+        protocol = 2
+        return cPickle.dumps(obj, protocol)
+
+    @staticmethod
+    def loads(obj):
+        return cPickle.loads(obj)
+
+    @classmethod
+    def pickle_and_write(cls, obj, stream):
+        serialized = cls.dumps(obj)
+
+        if serialized is None:
+            raise ValueError("Serialized value should not be None.")
+        elif len(serialized) > (1 << 31):
+            raise ValueError("Cannot serialize object larger than 2G.")
+
+        cls.write_int(len(serialized), stream)
+        stream.write(serialized)
+
+
+def main(in_file, out_file):
+    pickle_serializer = PickleSerializer()
+    try:
+        func = pickle_serializer.read_command(in_file)
+        tuple_size = pickle_serializer.read_int(in_file)
+        output_type = pickle_serializer.read_int(in_file)
+        is_flatmap = pickle_serializer.read_int(in_file)
+
+        if tuple_size < 1:
+            raise ValueError("Size of tuple should not be less than 1.")
+
+        while True:
+            num_tuples = pickle_serializer.read_int(in_file)
+            if num_tuples == SpecialLengths.END_OF_STREAM:
+                break
+
+            tuple_list = []
+            for j in range(num_tuples):
+                tuple_list.append(pickle_serializer.read_tuple(in_file, tuple_size))
+
+            retval = func(tuple_list)
+            if is_flatmap > 0:
```

Comment:
Please change to `if is_flatmap:`, since `is_flatmap` is implicitly boolean (can only take values in `{0,1}` per your format spec).